### PR TITLE
SysHealth: remove stale storage listeners

### PR DESCRIPTION
**Pull request by ianull**
This fixes the issue #4475 completely.

There are two leakage paths:
a) particle re-instantiation doesn't deregister stale storage listeners
b) Arc tear-down process doesn't deregister stale storage listeners (listening to any of mapped 
    stores.

The zombie storage listeners would cause:
a) memory leakage
b) regressed latency over time as iterating all registered (and increasing) listeners
c) unexpected behavior or crash on ingestion Arcs (old, aka. prior-to-reinstantiation callbacks get called surprisingly).

Note:
I don't exercise on StorageNG yet on device (still waiting), so put an TODO on its stack.

### CODE REVIEW

File: src/runtime/particle-execution-host.ts [hunk 1]
```diff
@@ -57,7 +57,10 @@ export class ParticleExecutionHost {
 
   constructor({slotComposer, arc, ports}: ParticleExecutionHostOptions) {
     this.close = () => {
-      this._apiPorts.forEach(apiPort => apiPort.close());
+      this._apiPorts.forEach(apiPort => {
```
Comment (sjmiles) [inline src/runtime/particle-execution-host.ts]: Suggest we make a stub for `clear()` in APIPort, and then call `clear()` inside of `close()`. Then we can avoid the typecast and simplify the calling code here in ParticleExecutionHost.

**Comment by sjmiles** [comment]
Thanks for jumping on this issue. Nice work.

I started looking at this last week, but I started chasing some other leaks that I found in other areas, so you beat me to this specific problem.

If you don't want to make the change I suggested, just let me know and we can figure out alternatives.

File: src/runtime/particle-execution-host.ts (hunk 1 above)
Comment (ianull) [reply src/runtime/particle-execution-host.ts]: Good suggestion, thank you.
Added a stub in APIPort and remove all type casts!

Comment (ianull) [reply src/runtime/particle-execution-host.ts]: Also called clear inside close.

Comment (ianull) [reply src/runtime/particle-execution-host.ts]: Done!

**Comment by ianull** [comment]
Welcome!

**Pull request merged**